presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -330,7 +336,9 @@ public static ColumnMapping timeColumnMapping(ConnectorSession session)
                     TIME,
                     (resultSet, columnIndex) -> {
                         LocalTime time = resultSet.getObject(columnIndex, LocalTime.class);
-                        return toPrestoLegacyTimestamp(time.atDate(LocalDate.ofEpochDay(0)), sessionZone);
+                        // disable rounding for time
```

Comment:
Presto stores LocaTime of 23:59:59.999, then reads LocalTime 23:59:59.999999, rounding rounds to 00:00:00. 
This is coherent with `NANOSECONDS.toMillis(time.toNanoOfDay());` of non legacy time.

Reply:
Ah yes, I ran into a similar issue when implementing variable precision time type: https://github.com/prestosql/presto/pull/4381. That's why I had to add `timeColumnMappingWithTruncation` for connectors that need those semantics. FYI, we're removing legacy time in that PR. Also, some of these changes will conflict with that PR, since you're modifying the parts that relate to TIME. I would leave those alone for now, since they are not related to TIMESTAMP(6).

Reply:
Does this fix a bug that we already have?
Or, we would have?